### PR TITLE
[Part 2] Re-factor publishing charts - gh-pages

### PR DESCRIPTION
**Pull request by ljubon**
The goal of this PR is to separate `gh-pages` and source code a.k.a `charts`

- `master` branch
  - place for source code 🙏 (e.g charts, values..etc)
- `gh-pages` branch
  - all public relates stuff (e.g `index.yaml` for repo, raw chart files...etc)

_NOTE_: 2nd PR https://github.com/G-Research/charts/pull/38

### CODE REVIEW

**Comment by ljubon** [comment]
add latest changes from upstream -> https://github.com/G-Research/charts/pull/7
`gh-pages` is up to date with upstream now

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+      - name: Update repo index 
+        run: |
+          echo "${{ secrets.CLONE_KEY }}" > id_rsa && chmod 400 id_rsa
+          trap "rm -f id_rsa" EXIT
+
+          helm repo index .
+          git diff index.yaml
+          git add index.yaml
+          git -c user.name="Ljubo Nikolic" -c user.email="ljubonikolic@gmail.com" commit -qam "Updating index.yaml"
```
Comment (76creates) [inline .github/workflows/gh-pages.yml:32]: This will need change as well I assume?

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          ssh-agent bash -c 'ssh-add id_rsa; git push -q origin HEAD'
+
+  update-armada-charts:
+    if: ${{ github.event.inputs.project == 'ljubon/armada' }}
```
Comment (76creates) [inline .github/workflows/gh-pages.yml:36]: ditto

**Comment by 76creates** [comment]
Im failing to see how generating `index.yaml` changes anything for [charts GH page](https://g-research.github.io/charts/) yet its a part of new and extended CI? 
Looks to me like that besides splitting we are adding some new features as well? Please be bit more verbose on this matter if so, thanks! ⚡

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (ljubon) [reply .github/workflows/gh-pages.yml:32]: my creds should be replaced by repo owners, so we will wait for them to review these changes and then action will be required by them to add official repo creds 🙏

File: .github/workflows/gh-pages.yml (hunk 2 above)
Comment (ljubon) [reply .github/workflows/gh-pages.yml:36]: i assume it's same question as above :D

**Comment by ljubon** [comment]
> Im failing to see how generating `index.yaml` changes anything for [charts GH page](https://g-research.github.io/charts/) yet its a part of new and extended CI?
> Looks to me like that besides splitting we are adding some new features as well? Please be bit more verbose on this matter if so, thanks! ⚡

hey @76creates I just saw your last comment, sorry for the late reply 🙏 
generating (or updating) `index.yaml` is same as before `helm repo index .` 
what is shown in the diff for `index.yaml` is just the latest timestamp i did update `index.yaml` with current charts
as soon as PR is merged, it will get latest/current state of gh-pages and `index.yaml` will be latest for helm repo

p.s a bit of background ~ `index.yaml` file should be updated with each new chart published to `gh-pages` since that file helm repo (public/remote) is watching when they want to reference our new charts

**Comment by 76creates** [comment]
@ljubon sorry I wasn't clear enough I think, just wanted to make sure that those changes comply with what you described in the PR which is:
```
The goal of this PR is to separate gh-pages and source code a.k.a charts
```

if so, please carry on, if not just add it to the PR description as well please ⚡  ty

**Comment by ljubon** [comment]
ping ping 👀

**Comment by ljubon** [comment]
Moving this PR to clean (non-gh-paged) due to dependacy over Armada PR

**Pull request closed without merging**